File: 2023/08/part1.cpp
#include <bits/stdc++.h>
using namespace std;

map<string, pair<string, string>> graph;
string inst;

int wander(const string& curr, int i = 0) {
    if (curr == "ZZZ") return 0;
    if (i == inst.size()) i = 0;
    if (inst[i] == 'L') {
        return 1 + wander(graph[curr].first, i + 1);
    } else {
        return 1 + wander(graph[curr].second, i + 1);
    }
}

int main() {
    ifstream f("input");
    string curr, left, right;
    f >> inst;
    while (f >> curr >> left >> left >> right) {
        graph[curr].first = left.substr(1, 3);
        graph[curr].second = right.substr(0, 3);
    }
    cout << wander("AAA");
}

File: 2023/08/part1.jl
f = open("input", "r")
inst = readline(f)
readline(f)
g = Dict()
for l in eachline(f)
    g[l[1:3]] = (l[8:10] l[13:15])
end
(p1 = 0; i = 1; s = "AAA")
while (s != "ZZZ")
    global p1 += 1
    global s = g[s][inst[i % (length(inst) + 1)] == 'L' ? 1 : 2]
    i += 1
end
println(p1)